From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fgei <fgei@gmail.com>
Date: Wed, 10 Jul 2024 13:47:10 +0000
Subject: [PATCH] tmp: config-generator: Backport WebView context leak fix
 upstream

Original commit message:

webview: disable Device Posture API due to leak.

The listener registration required for the Device Posture API leaks a
reference to the WebView's host Context to a GC root in the Android
framework, which means that apps that neglect to call WebView.destroy()
will leak the WebView and their Activity, potentially leaking huge
amounts of memory.

WebView promises not to leak in this scenario (due to the very high
prevalence of apps that don't call destroy()) so until a better solution
is found, disable the API in WebView.
---
 .../vanadium/config/host/ConfigGenerator.java    | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/vanadium/android_config/proto/java/src_host/app/vanadium/config/host/ConfigGenerator.java b/vanadium/android_config/proto/java/src_host/app/vanadium/config/host/ConfigGenerator.java
index 14713fcd5565b..7f1c4cfbc9d95 100644
--- a/vanadium/android_config/proto/java/src_host/app/vanadium/config/host/ConfigGenerator.java
+++ b/vanadium/android_config/proto/java/src_host/app/vanadium/config/host/ConfigGenerator.java
@@ -55,6 +55,22 @@ public class ConfigGenerator {
                         )
                 ))
         ));
+        configList.add(config(configParams -> configParams.setSpec(
+                        spec(specParams -> specParams.setSpecTypes(getSpecTypes(
+                                SpecType.WEBVIEW
+                        ))
+                                .setMinVersion(6478_000_00L)
+                                .setMaxVersion(6478_174_99L)
+                        ))
+                .addAllFlags(flags(
+                        flag(flagParams -> flagParams.setFlagName("DevicePosture")
+                                .setFlagType(FlagType.DISABLED_FEATURE)
+                        ),
+                        flag(flagParams -> flagParams.setFlagName("ViewportSegments")
+                                .setFlagType(FlagType.DISABLED_FEATURE)
+                        )
+                ))
+        ));
         configList.add(config(configParams -> configParams.setSpec(
                         spec(specParams -> specParams.setSpecTypes(getSpecTypes(
                                 SpecType.BROWSER
